src/CMakeLists.txt

```diff
@@ -9,6 +9,6 @@ set(SOURCES
 set(CMAKE_EXE_LINKER_FLAGS "-shared -fPIC ${CMAKE_EXE_LINKER_FLAGS}")
 
 add_library(${PROJECT_NAME} SHARED ${SOURCES})
-target_link_libraries(${PROJECT_NAME} -ldl)
+target_link_libraries(${PROJECT_NAME} -ldl ${Boost_LIBRARIES})
```

Comment:
Property Tree is a header-only library, so you don't need to link yebash with Boost libs